src/utils.rs

```diff
@@ -0,0 +1,190 @@
+    Ok(x)
+}
+
+/// It updates the register COND where we have the condition flag
+pub fn update_flags(r: Register, regs: &mut Registers) {
+    if regs[r] == 0 {
+        regs[Register::Cond] = CondFlag::Zro.value();
+    } else if regs[r] >> 15 == 1 {
+        regs[Register::Cond] = CondFlag::Neg.value();
+    } else {
+        regs[Register::Cond] = CondFlag::Pos.value();
+    }
+}
+
+/// Reads one byte from the standard input
+pub fn getchar(reader: &mut impl Read) -> Result<[u8; 1], VMError> {
+    let mut buffer = [0u8; 1];
+    reader
+        .read_exact(&mut buffer)
+        .map_err(|e: Error| VMError::STDINRead(e.to_string()))?;
+    Ok(buffer)
+}
+
+/// Flushes the writer
+///
+/// ### Returns
+///
+/// A Result indicating if the flushing succeded or not
+pub fn stdout_flush(writer: &mut impl Write) -> Result<(), VMError> {
+    writer
+        .flush()
+        .map_err(|_| VMError::STDOUTFlush(String::from("Cannot flush stdout")))?;
+    Ok(())
+}
+
+/// Writes the buffer into the writer
+///
+/// ### Returns
+///
+/// A Result indicating if the writting succeded or not
+pub fn stdout_write(buffer: &[u8], writer: &mut impl Write) -> Result<(), VMError> {
+    writer
+        .write_all(buffer)
+        .map_err(|_| VMError::STDOUTWrite(String::from("Cannot write on stdout")))?;
+    Ok(())
+}
+
+/// Sets the handling for the SIGINT signal and
+/// disables the input buffering on the terminal
+pub fn setup() -> Result<Termios, VMError> {
+    disable_input_buffering()
+}
+
+/// Restores the termios to the original one
+pub fn shutdown(initial_termios: Termios) -> Result<(), VMError> {
+    let stdin_fd = stdin().lock().as_raw_fd();
+    tcsetattr(stdin_fd, TCSANOW, &initial_termios).map_err(|_| {
+        VMError::TermiosSetup(String::from("Cannot set termios when shutting down"))
+    })?;
+    Ok(())
+}
+
+/// Gets the initial termios and disables its input buffering
+fn disable_input_buffering() -> Result<Termios, VMError> {
+    let stdin_fd = stdin().lock().as_raw_fd();
+    let initial_termios = Termios::from_fd(stdin_fd)
+        .map_err(|_| VMError::TermiosCreation(String::from("Cannot create termios")))?;
+    let mut new_termios = initial_termios;
+    new_termios.c_lflag &= !ICANON & !ECHO;
+    tcsetattr(stdin_fd, TCSANOW, &new_termios).map_err(|_| {
+        VMError::TermiosSetup(String::from("Cannot set termios with new attributes"))
+    })?;
+    Ok(initial_termios)
+}
+
+/// Load the file into the vm memory
+pub fn load_arguments(args: &mut Args, mem: &mut Memory) -> Result<(), VMError> {
+    if args.len() < 2 {
+        println!("lc3 [image-file1] ...");
+        exit(2);
+    }
+    // We skip the first element of the args since it is not an image
+    args.next();
+    for path in args {
+        if read_image(path.clone(), mem).is_err() {
+            println!("failed to load image: {path}");
+            exit(1);
+        }
+    }
+    Ok(())
+}
+
+/// Read bytes from file and send them to get into memory
```

Comment:
To keep consistency:

```suggestion
/// Reads bytes from file and send them to get into memory
```